src/components/Game.js

```diff
@@ -1,37 +1,200 @@
+    }
+
+    console.log('called country: ', randomCountry.name)
+
+    this.setState({
+      currentLat: latNum + latDir,
+      currentLong: longNum + longDir,
+      calledCoordinateIds: [...this.state.calledCoordinateIds, randomCountry.id]
+    })
+  }
+
+  setFilteredCountries() {
+    let filterRegion = this.props.region
+
+    if (this.props.region === 'australia-oceania') {
+      filterRegion = 'oceania'
+    } else if (this.props.region === 'north-america') {
+      filterRegion = 'north america'
+    } else if (this.props.region === 'south-america') {
+      filterRegion = 'south america'
+    }
+
+    let filteredCountries = this.state.allCountries.filter(country => {
+      let countryRegions = country.regions.data;
+      let lowerCaseRegions = countryRegions.map(region => {
+        return region.toLowerCase()
+      })
+      return lowerCaseRegions.includes(filterRegion)
+    })
+
+    if (this.props.region === 'world') {
+      filteredCountries = this.state.allCountries
+    }
+
+    this.setState({
+      currentRegionCountries: filteredCountries
+    })
+  }
+
+  handleBingoClick() {
+    this.state.bingoBtnTxt === 'Set Game Board!' ? this.setGameBoard() : this.evaluateBoard()
+  }
+
+  setGameBoard() {    
+    let randomCountries = []
+
+    while (randomCountries.length < 16) {
+      let randomIndex = getRandomIndex(this.state.currentRegionCountries)
+      let randomCountry = this.state.currentRegionCountries[randomIndex]
+      !randomCountries.includes(randomCountry) && randomCountries.push(randomCountry)
+    }
+
+    let squareCountries = randomCountries.map((randomCountry, index) => {
+      return {
+        name: randomCountry.name,
+        id: randomCountry.id,
+        flag_path: randomCountry.flag_path,
+        space: index,
+        isStamped: false,
+        correct: null
       }
+    })
+
+    this.setState({
+      currentBoard: squareCountries,
+      bingoBtnTxt: 'BINGO!'
+    })
+  }
+
+  evaluateBoard() {
+    let stampedSquares = this.state.currentBoard.filter(square => {
+      return square.isStamped
+    }).map(square => {
+      return square.space
+    })
+  
+    let wins = []
+    winningBoards.forEach(winBoard => {
+      let match = []
+      winBoard.forEach(square => {
+        match.push(stampedSquares.includes(square))
+      })
+      !match.includes(false) && wins.push(winBoard)
+    })
+
+    if (!wins.length) {
+      this.setState({winStatus: 'Keep playing - try to get four in a row!'})
+    } else {
+      let errors = 0
+        wins.forEach(win => {
+          win.forEach(square => {
+            if (this.state.calledCoordinateIds.includes(this.state.currentBoard[square].id)) {
+              this.setState(prevState => {
+                return prevState.currentBoard[square].correct = true
+              })
+            } else {
+              errors += 1
+              this.setState({winStatus: 'Oops! Not all of those countries were called - check the board for details'})
```

Comment:
Does this account for both error possibilities -- that the user incorrectly selected uncalled squares AND that they may have neglected to select squares that were called? (And do we care?)